### PR TITLE
fix link to subnet tagging in development guide

### PR DESCRIPTION
**Pull request by geoffcline**
**Issue, if available:**
in this guide https://karpenter.sh/docs/development-guide/ and the link embedded here (below) does not work:

> If you created your cluster on version 1.19 or above, you may need to tag your subnets as mentioned here.

**Description of changes:**
I changed the link to use a hugo reference, instead of linking to the old github content. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/docs/deploy-guide.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+title: "Development Guide"
```
Comment (ellistarn) [inline website/content/en/docs/deploy-guide.md:2]: Can we call this Operational Requirements?

File: website/content/en/docs/development-guide.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ make delete # Uninstall Karpenter
     * Make sure you have valid credentials to your development repository.
     * `$KO_DOCKER_REPO` must point to your development repository
     * Your cluster must have permissions to read from the repository
-* If you created your cluster on version 1.19 or above, you may need to tag your subnets as mentioned [here](docs/aws/README.md). This is a temporary problem with our subnet discovery system, and is being tracked [here](https://github.com/awslabs/karpenter/issues/404#issuecomment-845283904).
+* If you created your cluster on version 1.19 or above, you may need to tag your subnets as mentioned [here]({{< ref "{{< ref "/docs/getting-started/_index.md#tag-subnets" >}}" >}}). This is a temporary problem with our subnet discovery system, and is being tracked [here](https://github.com/awslabs/karpenter/issues/404#issuecomment-845283904).
```
Comment (ellistarn) [inline website/content/en/docs/development-guide.md:37]: Curious why you need a special ref. Don't markdown links work?

File: website/content/en/docs/deploy-guide.md (hunk 1 above)
Comment (ellistarn) [reply website/content/en/docs/deploy-guide.md:2]: Guessing you didn't mean to commit this.

Comment (geoffcline) [reply website/content/en/docs/deploy-guide.md]: yes, this has been removed from the PR :)

File: website/content/en/docs/development-guide.md (hunk 2 above)
Comment (geoffcline) [reply website/content/en/docs/development-guide.md]: I *think* it's necessary when you are using an "#" anchor on a "_index.md"

**Comment by geoffcline** [comment]
sorry for the PR that should have been a draft 😅 

it's better now

**Review by ellistarn — APPROVED** [review]

**Pull request merged**